main.cpp

```diff
@@ -19,13 +19,30 @@ using namespace clang::tooling;
 
 class CastCallBack : public MatchFinder::MatchCallback {
 public:
-    CastCallBack(Rewriter& rewriter) {
-        // Your code goes here
-    };
-
+    CastCallBack(Rewriter& rewriter): _rewriter(rewriter) {};
+    
     void run(const MatchFinder::MatchResult &Result) override {
-        // Your code goes here
+        auto *cast = Result.Nodes.getNodeAs<CStyleCastExpr>("cast");

```

Comment:
What if `cast` is a null pointer? How to protect the program from crashing in such cases?

Reply:
To protect our program from crash we need to add this "if statement" :
```c++
if(auto *cast = Result.Nodes.getNodeAs<CStyleCastExpr>("cast"))
{
	// same code as it was
}
```